src/energyplus/ForwardTranslator/ForwardTranslateHeatPumpAirToWaterFuelFiredCooling.cpp

```diff
@@ -0,0 +1,210 @@
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include "../ForwardTranslator.hpp"
+#include "../../model/Model.hpp"
+
+#include "../../model/HeatPumpAirToWaterFuelFiredCooling.hpp"
+#include "../../model/Node.hpp"
+#include "../../model/Node_Impl.hpp"
+#include "../../model/HeatPumpAirToWaterFuelFiredHeating.hpp"
+#include "../../model/HeatPumpAirToWaterFuelFiredHeating_Impl.hpp"
+#include "../../model/CurveBiquadratic.hpp"
+#include "../../model/CurveQuadratic.hpp"
+
+#include <utilities/idd/HeatPump_AirToWater_FuelFired_Cooling_FieldEnums.hxx>
+#include "../../utilities/idd/IddEnums.hpp"
+#include <utilities/idd/IddEnums.hxx>
+
+using namespace openstudio::model;
+
+namespace openstudio {
+
+namespace energyplus {
+
+  boost::optional<IdfObject> ForwardTranslator::translateHeatPumpAirToWaterFuelFiredCooling(model::HeatPumpAirToWaterFuelFiredCooling& modelObject) {
+    boost::optional<IdfObject> result;
+    boost::optional<WorkspaceObject> _wo;
+    boost::optional<ModelObject> _mo;
+
+    // Instantiate an IdfObject of the class to store the values
+    IdfObject idfObject = createRegisterAndNameIdfObject(openstudio::IddObjectType::HeatPump_AirToWater_FuelFired_Cooling, modelObject);
+
+    // Water Inlet Node Name
+    _mo = modelObject.inletModelObject();
+    if (_mo) {
+      translateAndMapModelObject(*_mo);
+      idfObject.setString(openstudio::HeatPump_AirToWater_FuelFired_CoolingFields::WaterInletNodeName, _mo->nameString());
+    }
+
+    // Water Outlet Node Name
+    _mo = modelObject.outletModelObject();
+    if (_mo) {
+      translateAndMapModelObject(*_mo);
+      idfObject.setString(openstudio::HeatPump_AirToWater_FuelFired_CoolingFields::WaterOutletNodeName, _mo->nameString());
+    }
+
+    // Air Source Node Name
+    std::string oaNodeName = modelObject.nameString() + " OA Node";
+    idfObject.setString(HeatPump_AirToWater_FuelFired_CoolingFields::AirSourceNodeName, oaNodeName);
+
+    IdfObject _oaNode(openstudio::IddObjectType::OutdoorAir_Node);
+    _oaNode.setString(0, oaNodeName);
+    m_idfObjects.push_back(_oaNode);
+
+    // Companion Heating Heat Pump Name: Optional Object
+    if (boost::optional<HeatPumpAirToWaterFuelFiredHeating> _companionHeatingHeatPump = modelObject.companionHeatingHeatPump()) {
+      if (boost::optional<IdfObject> _owo = translateAndMapModelObject(_companionHeatingHeatPump.get())) {
+        idfObject.setString(HeatPump_AirToWater_FuelFired_CoolingFields::CompanionHeatingHeatPumpName, _owo->nameString());
+      }
+    }
```

Comment:
Ok, you're using createRegisterAndNameIdfObject so that should be fine. Otherwise we'd be in an infinite loop.